From 9d0ee3d2dc1af6c2d28d7da84ccb187ba9d06e2a Mon Sep 17 00:00:00 2001
From: Jakub Pavlik <pavlk.jakub@gmail.com>
Date: Tue, 3 Nov 2015 10:34:45 +0100
Subject: [PATCH] Fix import fixes for compatability with OpenStack Liberty
 Release. Create noop methods for Contrail quota driver to support Neutron
 reservation in Liberty.

Change-Id: I1f47c980ea65c32753a7f0038403299c7b76c47b
Closes-Bug: #1508467
---
 neutron_plugin_contrail/extensions/ipam.py         |  5 +++-
 neutron_plugin_contrail/extensions/policy.py       |  5 +++-
 .../extensions/vpcroutetable.py                    | 10 ++++++--
 .../opencontrail/loadbalancer/loadbalancer_db.py   |  8 +++++-
 .../loadbalancer/loadbalancer_healthmonitor.py     |  6 ++++-
 .../opencontrail/loadbalancer/loadbalancer_pool.py |  6 ++++-
 .../opencontrail/loadbalancer/virtual_ip.py        |  6 ++++-
 .../plugins/opencontrail/quota/driver.py           | 30 +++++++++++++++++++++-
 8 files changed, 67 insertions(+), 9 deletions(-)

diff --git a/neutron_plugin_contrail/extensions/ipam.py b/neutron_plugin_contrail/extensions/ipam.py
index 747efe6..735d183 100644
--- a/neutron_plugin_contrail/extensions/ipam.py
+++ b/neutron_plugin_contrail/extensions/ipam.py
@@ -5,8 +5,11 @@ from neutron.api.v2 import base
 from neutron.common import exceptions as qexception
 from neutron.api import extensions
 from neutron import manager
-from oslo.config import cfg
 
+try:
+    from oslo.config import cfg
+except ImportError:
+    from oslo_config import cfg
 
 # Ipam Exceptions
 class IpamNotFound(qexception.NotFound):
diff --git a/neutron_plugin_contrail/extensions/policy.py b/neutron_plugin_contrail/extensions/policy.py
index 418969b..85ecffd 100644
--- a/neutron_plugin_contrail/extensions/policy.py
+++ b/neutron_plugin_contrail/extensions/policy.py
@@ -5,8 +5,11 @@ from neutron.api.v2 import base
 from neutron.common import exceptions as qexception
 from neutron.api import extensions
 from neutron import manager
-from oslo.config import cfg
 
+try:
+    from oslo.config import cfg
+except ImportError:
+    from oslo_config import cfg
 
 # Policy Exceptions
 class PolicyNotFound(qexception.NotFound):
diff --git a/neutron_plugin_contrail/extensions/vpcroutetable.py b/neutron_plugin_contrail/extensions/vpcroutetable.py
index 776c6e8..19edcd7 100644
--- a/neutron_plugin_contrail/extensions/vpcroutetable.py
+++ b/neutron_plugin_contrail/extensions/vpcroutetable.py
@@ -18,16 +18,22 @@
 from abc import ABCMeta
 from abc import abstractmethod
 
-from oslo.config import cfg
+try:
+    from oslo.config import cfg
+except ImportError:
+    from oslo_config import cfg
 
 from neutron.api import extensions
 from neutron.api.v2 import attributes as attr
 from neutron.api.v2 import base
 from neutron.common import exceptions as qexception
 from neutron import manager
-from neutron.openstack.common import uuidutils
 from neutron import quota
 
+try:
+    from neutron.openstack.common import uuidutils
+except ImportError:
+    from oslo_utils import uuidutils
 
 # Route table Exceptions
 class RouteTableNotFound(qexception.NotFound):
diff --git a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_db.py b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_db.py
index 2c9828f..614163b 100644
--- a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_db.py
+++ b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_db.py
@@ -6,7 +6,13 @@ import requests
 import time
 import uuid
 
-from neutron.common.config import cfg
+try:
+    from neutron.common.config import cfg
+except ImportError:
+    try:
+        from oslo.config import cfg
+    except ImportError:
+        from oslo_config import cfg
 
 from cfgm_common import analytics_client
 from cfgm_common import exceptions as vnc_exc
diff --git a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_healthmonitor.py b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_healthmonitor.py
index eadebcb..82639ab 100644
--- a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_healthmonitor.py
+++ b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_healthmonitor.py
@@ -9,7 +9,11 @@ try:
 except ImportError:
     from neutron_lbaas.extensions import loadbalancer
 
-from neutron.openstack.common import uuidutils
+try:
+    from neutron.openstack.common import uuidutils
+except ImportError:
+    from oslo_utils import uuidutils
+
 from vnc_api.vnc_api import IdPermsType
 from vnc_api.vnc_api import LoadbalancerHealthmonitor
 from vnc_api.vnc_api import LoadbalancerHealthmonitorType
diff --git a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_pool.py b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_pool.py
index af894ef..60956f5 100644
--- a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_pool.py
+++ b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/loadbalancer_pool.py
@@ -13,7 +13,11 @@ from neutron.api.v2 import attributes as attr
 from neutron.plugins.common import constants
 from neutron.services import provider_configuration as pconf
 
-from neutron.openstack.common import uuidutils
+try:
+    from neutron.openstack.common import uuidutils
+except ImportError:
+    from oslo_utils import uuidutils
+
 from vnc_api.vnc_api import *
 
 from resource_manager import ResourceManager
diff --git a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/virtual_ip.py b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/virtual_ip.py
index 75f915f..65d482e 100644
--- a/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/virtual_ip.py
+++ b/neutron_plugin_contrail/plugins/opencontrail/loadbalancer/virtual_ip.py
@@ -15,7 +15,11 @@ try:
 except ImportError:
     from oslo_log import log as logging
 
-from neutron.openstack.common import uuidutils
+try:
+    from neutron.openstack.common import uuidutils
+except ImportError:
+    from oslo_utils import uuidutils
+
 from vnc_api.vnc_api import IdPermsType, NoIdError
 from vnc_api.vnc_api import InstanceIp, VirtualMachineInterface
 from vnc_api.vnc_api import SecurityGroup
diff --git a/neutron_plugin_contrail/plugins/opencontrail/quota/driver.py b/neutron_plugin_contrail/plugins/opencontrail/quota/driver.py
index 5db7060..e3b7928 100644
--- a/neutron_plugin_contrail/plugins/opencontrail/quota/driver.py	2015-12-07 00:27:22.269936335 -0800
+++ b/neutron_plugin_contrail/plugins/opencontrail/quota/driver.py	2015-12-07 00:38:19.769966750 -0800
@@ -8,7 +8,14 @@
 except ImportError:
     from oslo_log import log as logging

-from neutron.common.config import cfg
+try:
+    from neutron.common.config import cfg
+except ImportError:
+    try:
+        from oslo.config import cfg
+    except ImportError:
+        from oslo_config import cfg
+
 from neutron.common import exceptions
 from httplib2 import Http
 import re
@@ -21,6 +28,12 @@
 from cfgm_common import exceptions as vnc_exc
 from vnc_api import vnc_api

+try:
+    from neutron.db.quota import api as quota_api
+except ImportError:
+    pass
+
+
 LOG = logging.getLogger(__name__)

 vnc_conn = None
@@ -47,6 +60,22 @@
             'health_monitor': 'loadbalancer_healthmonitor'
             };

+    def make_reservation(self, context, tenant_id, resources, deltas, plugin):
+        """This driver does not support reservations.
+
+        This routine is provided for backward compatibility purposes with
+        the API controllers which have now been adapted to make reservations
+        rather than counting resources and checking limits - as this
+        routine ultimately does.
+        """
+        return quota_api.ReservationInfo('fake', None, None, None)
+
+    def commit_reservation(self, context, reservation_id):
+        """Tnis is a noop as this driver does not support reservations."""
+
+    def cancel_reservation(self, context, reservation_id):
+        """Tnis is a noop as this driver does not support reservations."""
+
     @classmethod
     def _get_vnc_conn(cls):
         global vnc_conn
-- 
1.9.1

